meson.build

```diff
@@ -117,7 +117,7 @@ if sm_check.stderr() != ''
     warning('Submodule check completed with errors:\n@0@'.format(sm_check.stderr()))
 endif
 
-static = get_option('static') or (host_machine.system() == 'emscripten')
+static = get_option('static') or (host_machine.system() in ['emscripten', 'nx'])
```

Comment:
The `x in list` syntax was added in meson 0.49.0; unfortunately we still need to support at least 0.46.0 to some extent. It's ok if the switch port won't build on it, but the file should at least parse.

Changing it to `['emscripten', 'nx'].contains(host_machine.system())` should fix the issue.